src/Agent.Worker/TaskManager.cs

```diff
@@ -410,6 +411,68 @@ public sealed class Definition
         public DefinitionData Data { get; set; }
         public string Directory { get; set; }
         public string ZipPath { get; set; }
+
+        public TaskVersion GetPowerShellSDKVersion()
+        {
+            var modulePath = Path.Combine(Directory, "ps_modules", "VstsTaskSdk", "VstsTaskSdk.psd1");
+            if (!File.Exists(modulePath))
+            {
+                return null;
+            }
+
+            var versionLine = File.ReadAllLines(modulePath).FirstOrDefault(x => x.Contains("ModuleVersion"));
+            if (string.IsNullOrEmpty(versionLine))
+            {
+                return null;
+            }
+
+            var verRegex = new Regex(@"\d+\.\d+\.\d+");
```

Comment:
>verRegex [](http://example.com/codeflow?start=16&length=8)

I'm thinking this goes a bit far to avoid exceptions in TaskVersion.  It's unfortunate that Task Version doesn't have a TryParse, but we can just use catch(ArgumentException) in this case.

At the same time, it also doesn't preview all cases where TaskVersion might throw an exception

Reply:
I know this logic might seem a bit overloaded. I implemented it this way not to avoid exceptions, but to ensure correct behavior.

Originally, the TaskVersion class was designed to handle versions of agent tasks, not package versions. I decided to repurpose it to avoid adding additional logic for version comparison (especially since it includes a necessary 'IsTest' flag).

So, if I directly input the '5.0.1-preview.0' version into the TaskVersion constructor method, an exception will be raised since the TaskVersion doesn't expect the word 'preview' or more than three delimiter signs ('.' or '-').

In short, this requires preprocessing.